internal/node/machine_mgr.go

```diff
@@ -547,12 +551,46 @@ func (m *MachineManager) cleanSockets() {
 
 func (m *MachineManager) generateTriggerHandler(vm *runningFirecracker, tsub string, request *agentapi.DeployRequest) func(msg *nats.Msg) {
 	return func(msg *nats.Msg) {
+		ctx := m.ctx
```

Comment:
If we drop this line and use `m.ctx` on line 557, I think it makes it a lot clearer.